integration/pkg/api/osrm/table/response.go

```diff
@@ -0,0 +1,23 @@
+package table
+
+import "github.com/Telenav/osrm-backend/integration/pkg/api/osrm/route"
+
+// Response represents OSRM api v1 table response.
+type Response struct {
+	Code         string         `json:"code"`
+	Message      string         `json:"message,omitempty"`
+	Sources      []*Source      `json:"sources"`
+	Destinations []*Destination `json:"destinations"`
+	Durations    [][]*float64   `json:"durations"`
+	Distances    [][]*float64   `json:"distances"`
+}
+
+// Source represents as way point object.  All sources will be listed in order.
+type Source struct {
+	route.Waypoint
```

Comment:
If the `Waypoint` shares between `table` and `route` service, let's put it outside `route`. I think just put the definition to the `osrm/` will be fine.

Reply:
Also, the extra `type` is not necessary. You can define it by `Sources []*Waypoint` directly.